pkg/controller/volume/scheduling/scheduler_binder.go

```diff
@@ -237,8 +255,9 @@ func (b *volumeBinder) FindPodVolumes(pod *v1.Pod, node *v1.Node) (unboundVolume
 		for _, claim := range claimsToBind {
 			if selectedNode, ok := claim.Annotations[pvutil.AnnSelectedNode]; ok {
 				if selectedNode != node.Name {
-					// Fast path, skip unmatched node
-					return false, boundVolumesSatisfied, nil
+					// Fast path, skip unmatched node.
+					unboundVolumesSatisfied = false
+					goto done
```

Comment:
I don't see goto used anywhere substantial in our codebase. How about using defer, and changing this back to return?

Reply:
So a `return nil, nil` here instead of `goto done` and then in a deferred function check if `err == nil` and if yes, then add the reason strings? I had considered that, but it seemed more complicated and I wasn't sure which approach would be considered better style. I can change it.

Reply:
Or better, just plain `return` - then it is obvious that return value handling happens elsewhere.